runtime/cudaq/algorithms/run.h

```diff
@@ -37,6 +43,47 @@ struct RunResultSpan {
 RunResultSpan runTheKernel(std::function<void()> &&kernel,
                            quantum_platform &platform,
                            const std::string &kernel_name, std::size_t shots);
+
+// Template to transfer the ownership of the buffer in a RunResultSpan to a
+// `std::vector<T>` object. This special code is required because a
+// `std::vector<T>` will always construct its own data, and own it, using its
+// standard constructors. In this case, we are transferring ownership of a
+// buffer to the vector, `result`, and do not want to make a copy.
+template <typename T>
+void resultSpanToVectorViaOwnership(std::vector<T> &result,
+                                    RunResultSpan &spanIn) {
+  using raw_vector = struct {
+    T *start;
+    T *end0;
+    T *end1;
+  };
+  static_assert(sizeof(std::vector<T>) == sizeof(raw_vector) &&
+                "std::vector must use the nominal 3 pointer implementation");
+
+  // Swap vec into a local variable. vec's original content, if any will be
+  // reclaimed at the end of this function.
+  std::vector<T> deadEnder;
+  std::swap(deadEnder, result);
```

Comment:
Would returning the local variable have the same effect (instead of passing a reference as an argument and swapping it with a local? I mean after the RVO?:)

Reply:
Yes. What I have here is manual RVO.